synapse/rest/client/v2_alpha/account.py

```diff
@@ -603,15 +617,10 @@ async def on_GET(self, request):
 
             # Perform a 302 redirect if next_link is set
             if next_link:
-                if next_link.startswith("file:///"):
-                    logger.warning(
-                        "Not redirecting to next_link as it is a local file: address"
-                    )
```

Comment:
This check now happens implicitly in `assert_valid_next_link`, as only `http` and `https` `next_link` values are allowed.

Additionally, `next_link` can only be specified during the call to `/requestToken`. Having this check in `/submit_token`, blocking the user after already approving their `/requestToken` call, is not great UX.